codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -301,7 +301,15 @@ public ByteBuf getChunk(int length) throws IOException {
         int read = 0;
         ByteBuffer byteBuffer = ByteBuffer.allocate(length);
         while (read < length) {
-            int readnow = fileChannel.read(byteBuffer);
+            int readnow=-1;
+            try {
+                readnow = fileChannel.read(byteBuffer);
+            }catch (IOException e){
+                logger.warn("Failed to read a file.", e);
+            }
+            finally {
+                fileChannel.close();
```

Comment:
Should this `try finally` better englobing the while loop? Because there, the read operation will be only valid once, then failed since it will be closed from the very first iteration...

Reply:
yes, according to your suggest. I rewrite the fix.